### PR TITLE
Fuse grid mean tendencies into single loop

### PR DESCRIPTION
**Pull request by charleskawczynski**
Looking through the code, I realized that _all_ grid mean quantities seem to be updated in a standardized way, and I think we can fuse all the tendencies into a single loop.

This is still not a great final version, but we can start thinking about how we want to dispatch to add sources for a given model.

### CODE REVIEW

**Review by yairchn — APPROVED** [review]
LGTM

File: src/Turbulence_PrognosticTKE.jl [hunk 1]
```diff
@@ -237,17 +237,82 @@ update_forcing(Case::CasesBase, GMV::GridMeanVariables, TS::TimeStepping) =
...
+        end
+
+        if force_type(Case.Fo) <: ForcingStandard
+            if Case.Fo.apply_subsidence
+                GMV.H.tendencies[k] -= ∇H * Case.Fo.subsidence[k]
+                GMV.QT.tendencies[k] -= ∇q_tot * Case.Fo.subsidence[k]
+            end
+            GMV.H.tendencies[k] += Case.Fo.dTdt[k] / Π
+            GMV.QT.tendencies[k] += Case.Fo.dqtdt[k]
+        end
```
Comment (haakon-e) [inline src/Turbulence_PrognosticTKE.jl:276]: A bit late to the party here, but isn't this kind of pattern ripe for multiple dispatch?
I'm assuming at most one of `force_type(Case.Fo) <: ...` should ever be applied

Comment (haakon-e) [reply src/Turbulence_PrognosticTKE.jl:276]: Oh, I see you mentioned this in the description. If we're returning to this later, LGTM for now.

Comment (charleskawczynski) [reply src/Turbulence_PrognosticTKE.jl:276]: Yes, definitely. That will be the long term solution. I figure that collecting all of these sources into a single loop is a good first step. Then we can dispatch on which sources to include (within the loop). This is basically how ClimateMachine did it

**Pull request merged**